? ifdefs.patch
? mobile
? wxNotebookBaseFit.wx_2_5.Patch
? samples/notebook/notebook.sln
? samples/notebook/notebook.suo
? samples/notebook/notebook.vcproj
? src/wxWindows.sln
? src/wxWindows.suo
? src/wxWindows.vcproj
? src/jpeg/jpeg.vcproj
? src/msw/moreifdefs.patch
? src/png/png.vcproj
? src/regex/regex.vcproj
? src/tiff/tiff.vcproj
? src/zlib/zlib.vcproj
Index: include/wx/notebook.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/notebook.h,v
retrieving revision 1.26
diff -b -u -2 -r1.26 notebook.h
--- include/wx/notebook.h	2003/04/02 11:10:29	1.26
+++ include/wx/notebook.h	2003/05/31 09:37:46
@@ -108,5 +108,5 @@
 
     // calculate the size of the notebook from the size of its page
-    virtual wxSize CalcSizeFromPage(const wxSize& sizePage);
+    virtual wxSize CalcSizeFromPage(const wxSize& sizePage) const;
 
     // operations
@@ -156,4 +156,6 @@
     // remove the page and return a pointer to it
     virtual wxNotebookPage *DoRemovePage(int page);
+	// return the minimum size large enough to display the largest page entirely
+	virtual wxSize DoGetBestSize() const;
 
     // common part of all ctors
Index: include/wx/mac/notebook.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/mac/notebook.h,v
retrieving revision 1.18
diff -b -u -2 -r1.18 notebook.h
--- include/wx/mac/notebook.h	2003/03/27 20:01:32	1.18
+++ include/wx/mac/notebook.h	2003/05/31 09:37:47
@@ -86,5 +86,5 @@
    
     // calculate size for wxNotebookSizer
-  wxSize CalcSizeFromPage(const wxSize& sizePage);
+  wxSize CalcSizeFromPage(const wxSize& sizePage) const;
 
 /*
Index: include/wx/msw/notebook.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/msw/notebook.h,v
retrieving revision 1.35
diff -b -u -2 -r1.35 notebook.h
--- include/wx/msw/notebook.h	2003/05/01 16:39:57	1.35
+++ include/wx/msw/notebook.h	2003/05/31 09:37:48
@@ -104,4 +104,7 @@
   void SetTabSize(const wxSize& sz);
 
+  // calculate the size of the notebook from the size of its page
+  virtual wxSize CalcSizeFromPage(const wxSize& sizePage) const;
+
   // callbacks
   // ---------
Index: include/wx/univ/notebook.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/include/wx/univ/notebook.h,v
retrieving revision 1.7
diff -b -u -2 -r1.7 notebook.h
--- include/wx/univ/notebook.h	2001/11/27 14:43:26	1.7
+++ include/wx/univ/notebook.h	2003/05/31 09:37:49
@@ -79,5 +79,5 @@
     virtual void SetTabSize(const wxSize& sz);
 
-    virtual wxSize CalcSizeFromPage(const wxSize& sizePage);
+    virtual wxSize CalcSizeFromPage(const wxSize& sizePage) const;
 
     virtual bool DeleteAllPages();
Index: src/common/nbkbase.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/common/nbkbase.cpp,v
retrieving revision 1.10
diff -b -u -2 -r1.10 nbkbase.cpp
--- src/common/nbkbase.cpp	2003/05/26 19:25:15	1.10
+++ src/common/nbkbase.cpp	2003/05/31 09:37:56
@@ -97,5 +97,5 @@
 // ----------------------------------------------------------------------------
 
-wxSize wxNotebookBase::CalcSizeFromPage(const wxSize& sizePage)
+wxSize wxNotebookBase::CalcSizeFromPage(const wxSize& sizePage) const
 {
     // this was just taken from wxNotebookSizer::CalcMin() and is, of
@@ -103,26 +103,9 @@
     wxSize sizeTotal = sizePage;
     
-    // Slightly less bogus, at least under Windows.
-    // We need to make getting tab size part of the wxWindows API.
-#ifdef __WXMSW__
-    wxSize tabSize(0, 0);
-    if (GetPageCount() > 0)
-    {
-        RECT rect;
-        TabCtrl_GetItemRect((HWND) GetHWND(), 0, & rect);
-        tabSize.x = rect.right - rect.left;
-        tabSize.y = rect.bottom - rect.top;
-    }
-    if ( HasFlag(wxNB_LEFT) || HasFlag(wxNB_RIGHT) )
-    {
-        sizeTotal.x += tabSize.x + 7;
-        sizeTotal.y += 7;
-    }
-    else
-    {
-        sizeTotal.x += 7;
-        sizeTotal.y += tabSize.y + 7;
-    }
-#else
+	// changed hajokirchhoff -- May, 31st, 2003
+	// moved the __WXMSW__ portion to wxNotebook::CalcSizeFromPage in src/msw/notebook.cpp
+	// where it really belongs.
+	// Question: Shouldn't we make wxNotebookBase::CalcSizeFromPage a pure virtual class.
+	//			 I'd like this better than this "totally bogus" code here.
     if ( HasFlag(wxNB_LEFT) || HasFlag(wxNB_RIGHT) )
     {
@@ -135,5 +118,4 @@
         sizeTotal.y += 40;
     }
-#endif
 
     return sizeTotal;
@@ -164,4 +146,19 @@
 
     return pageRemoved;
+}
+
+wxSize wxNotebookBase::DoGetBestSize() const
+{
+  wxSize bestSize(0,0);
+  size_t nCount = m_pages.Count();
+	// iterate over all pages, get the largest width and height
+  for ( size_t nPage = 0; nPage < nCount; nPage++ ) {
+    wxNotebookPage *pPage = m_pages[nPage];
+    wxSize childBestSize(pPage->GetBestSize());
+	bestSize.SetWidth(max(childBestSize.GetWidth(), bestSize.GetWidth()));
+	bestSize.SetHeight(max(childBestSize.GetHeight(), bestSize.GetHeight()));
+  }
+  // convert display area to window area, adding the size neccessary for the tab control itself
+  return CalcSizeFromPage(bestSize);
 }
 
Index: src/mac/notebmac.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/notebmac.cpp,v
retrieving revision 1.28
diff -b -u -2 -r1.28 notebmac.cpp
--- src/mac/notebmac.cpp	2003/05/29 20:30:54	1.28
+++ src/mac/notebmac.cpp	2003/05/31 09:37:58
@@ -203,5 +203,5 @@
 }
 
-wxSize wxNotebook::CalcSizeFromPage(const wxSize& sizePage)
+wxSize wxNotebook::CalcSizeFromPage(const wxSize& sizePage) const
 {
     wxSize sizeTotal = sizePage;
Index: src/msw/notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/notebook.cpp,v
retrieving revision 1.72
diff -b -u -2 -r1.72 notebook.cpp
--- src/msw/notebook.cpp	2003/05/23 19:16:44	1.72
+++ src/msw/notebook.cpp	2003/05/31 09:38:01
@@ -312,4 +312,31 @@
 }
 
+wxSize wxNotebook::CalcSizeFromPage(const wxSize& sizePage) const
+{
+    wxSize sizeTotal = sizePage;
+    
+    // We need to make getting tab size part of the wxWindows API.
+    wxSize tabSize(0, 0);
+    if (GetPageCount() > 0)
+    {
+        RECT rect;
+        TabCtrl_GetItemRect((HWND) GetHWND(), 0, & rect);
+        tabSize.x = rect.right - rect.left;
+        tabSize.y = rect.bottom - rect.top;
+    }
+    if ( HasFlag(wxNB_LEFT) || HasFlag(wxNB_RIGHT) )
+    {
+        sizeTotal.x += tabSize.x + 7;
+        sizeTotal.y += 7;
+    }
+    else
+    {
+        sizeTotal.x += 7;
+        sizeTotal.y += tabSize.y + 7;
+    }
+
+    return sizeTotal;
+}
+
 void wxNotebook::AdjustPageSize(wxNotebookPage *page)
 {
Index: src/univ/notebook.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/univ/notebook.cpp,v
retrieving revision 1.15
diff -b -u -2 -r1.15 notebook.cpp
--- src/univ/notebook.cpp	2002/12/04 14:10:28	1.15
+++ src/univ/notebook.cpp	2003/05/31 09:38:04
@@ -963,5 +963,5 @@
 }
 
-wxSize wxNotebook::CalcSizeFromPage(const wxSize& sizePage)
+wxSize wxNotebook::CalcSizeFromPage(const wxSize& sizePage) const
 {
     return AdjustSize(GetSizeForPage(sizePage));

 	  	 
